torch/csrc/PyInterpreter.cpp

```diff
@@ -322,6 +336,11 @@ void ConcretePyInterpreterVTable::dispatch(
             append_overloaded_tensor(&overloaded_args, py::cast(tensor).ptr());
           }
         }
+        if (nv.isSymInt()) {
+          if (nv.toSymNodeImplUnowned()->is_singleton()) {
+            append_overloaded_type(&overloaded_args, _get_nested_tensor_cls());
+          }
+        }
```

Comment:
To add as comment: The more generalized version of this, is we test if the SymNode has Python key set, and if so, we append_overloaded_object of the Python object bound in the singleton symint.

Reply:
Added